soc/nordic/common/pinctrl_soc.h

```diff
@@ -26,6 +27,18 @@ extern "C" {
 /** Type for nRF pin. */
 typedef uint32_t pinctrl_soc_pin_t;
 
+
+struct nrf_pinctrl_pin {
```

Comment:
Please move this struct and the hook declarations to a device driver specific header in include/zephyr/drivers/pinctrl/

these syms are not soc specific, they are device driver specific :) The SoC "merely" implements them :)

Reply:
Moved whole `pinctrl_soc.h` to `include`, to align with how other vendors.